src/libraries/System.Net.Quic/src/System/Net/Quic/Implementations/MsQuic/MsQuicConnection.cs

```diff
@@ -565,13 +568,34 @@ internal override ValueTask ConnectAsync(CancellationToken cancellationToken = d
             Debug.Assert(_state.StateGCHandle.IsAllocated);
 
             _state.Connection = this;
+            uint status;
+            if (_remoteEndPoint is IPEndPoint)
+            {
+                SOCKADDR_INET address = MsQuicAddressHelpers.IPEndPointToINet((IPEndPoint)_remoteEndPoint);
+                unsafe
+                {
+                    status = MsQuicApi.Api.SetParamDelegate(_state.Handle, QUIC_PARAM_LEVEL.CONNECTION, (uint)QUIC_PARAM_CONN.REMOTE_ADDRESS, (uint)sizeof(SOCKADDR_INET), (byte*)&address);
+                    QuicExceptionHelpers.ThrowIfFailed(status, "Failed to connect to peer.");
+                }
+
+                if (targetHost == null)
+                {
+                    targetHost = ((IPEndPoint)_remoteEndPoint).Address.ToString();
+                }
+            }
+            else
+            {
+                // We don't have way how to set separate SNI and name for connection at this moment.
+                targetHost = ((DnsEndPoint)_remoteEndPoint).Host;
+            }
+
             try
             {
-                uint status = MsQuicApi.Api.ConnectionStartDelegate(
+                status = MsQuicApi.Api.ConnectionStartDelegate(
```

Comment:
```suggestion
                uint status = MsQuicApi.Api.ConnectionStartDelegate(
```
It shouldn't clash if you put the declaration above next to the `SetParamDelegate` as well. They should be within different scopes.